src/main/java/com/maxmind/db/BufferHolder.java

```diff
@@ -52,21 +52,21 @@ final class BufferHolder {
         if (null == stream) {
             throw new NullPointerException("Unable to use a NULL InputStream");
         }
-        final int chunkSize = Integer.MAX_VALUE;
+        final int chunkSize = Integer.MAX_VALUE / 2;
```

Comment:
What was the motivation behind this change?

Reply:
Mostly cause of the changes in aafbae6. I made the same change in `MultiBuffer` too.

I was getting allocation errors trying to allocate `byte[Integer.MAX_VALUE]`, as far as I understood because when allocating an array some memory is reserved for various metadata. 

I noticed that `Integer.MAX_VALUE - 8` does the trick, at least on my machine, though I don't know if every platform would have been fine so I went with half max int to be safe.

Reply:
Actually, you should just set `DEFAULT_CHUNK_SIZE` to this and then use that here.